rosenpass/src/protocol.rs

```diff
@@ -1735,10 +1997,75 @@ impl CryptoServer {
 
         Ok(hs.peer())
     }
+
+    pub fn handle_cookie_reply(&mut self, cr: CookieReply<&[u8]>) -> Result<PeerPtr> {
```

Comment:
Lets avoid the super-deeply nested code here; there is a lot of rosenpass code demonstrating how `if ERROR { bail/return }` or `let VAR = match value {
 ERROR => bail,
 GOOD => value,
}` can be used to avoid unnecessarily deep indendation